### PR TITLE
Arkansas income tax rate

### LINKED ISSUE #2265: Arkansas tax rate
tax rate   income range                       tax due
0.0%        $0 - $5,099                        $0

2.0%.       $5,100 - $10,299.              $0 + 2.0% over $5,099.99
3.0%.       $10,300 - $14,699            $103 + 3.0% over $10,299.99
3.4%.       $14,700 - $24,299.99     $235.00 + 3.4% over $14,699.99
4.9%.       $24,300 - $87,000          $561 + 4.9% over $24,299.99

### PR DESCRIPTION
**Pull request by Carolinecc123**
### Summary
📝🧪💰

This pull request adds the functionality to calculate the Arkansas income tax before credits for 2022, based on the recent tax legislation. It includes the parameters, tests, and variable for the `ar_income_tax_before_credits` feature, as well as the relevant files in the `policyengine_us` package.

> _`calc` tax rates for_
> _Arkansas in twenty-two_
> _winter of changes_

### Walkthrough
*  Add Arkansas income tax rates for 2022 ([link](https://github.com/PolicyEngine/policyengine-us/pull/2297/files?diff=unified&w=0#diff-8ed7f30cff946945131ebd4ee1e3c1f4d7c87d33fe8ba2bc7e2275b94f8868b4R1-R33), [link](https://github.com/PolicyEngine/policyengine-us/pull/2297/files?diff=unified&w=0#diff-74ad8dd8698b2d28c4d415f6fd77ac85ea444880ba5e0ac612f579c2e351011bL1-R43), [link](https://github.com/PolicyEngine/policyengine-us/pull/2297/files?diff=unified&w=0#diff-c03f2543360263cca6a07683d6d33452b19100cf1e43711fa059e69291b6351dL1-R19))
  - Create `policyengine_us/parameters/gov/states/ar/tax/income/rates.yaml` with brackets, metadata, and references ([link](https://github.com/PolicyEngine/policyengine-us/pull/2297/files?diff=unified&w=0#diff-8ed7f30cff946945131ebd4ee1e3c1f4d7c87d33fe8ba2bc7e2275b94f8868b4R1-R33))
  - Define `ar_income_tax_before_credits` variable in `policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py` with value type, entity, label, unit, definition period, reference, state code, and formula ([link](https://github.com/PolicyEngine/policyengine-us/pull/2297/files?diff=unified&w=0#diff-c03f2543360263cca6a07683d6d33452b19100cf1e43711fa059e69291b6351dL1-R19))
  - Add tests for `ar_income_tax_before_credits` in `policyengine_us/tests/policy/baseline/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.yaml` with five test cases and expected output ([link](https://github.com/PolicyEngine/policyengine-us/pull/2297/files?diff=unified&w=0#diff-74ad8dd8698b2d28c4d415f6fd77ac85ea444880ba5e0ac612f579c2e351011bL1-R43))

fixes #2265

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ar/tax/income/main.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - title: State of Arkansas Estimated Tax Declaration Vouchers and Instructions for Tax Year 2023
+    href: https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/2023_Final_AR1000ES.pdf
+
+brackets:
+  - threshold:
+      2022-01-01: 0
+    rate:
+      2022-01-01: 0
+  - threshold:
+      2022-01-01: 5_099
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ar/tax/income/main.yaml:19]: I think it should be the round numbers

File: policyengine_us/parameters/gov/states/ar/tax/income/main.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+description: Arkansas taxes personal taxable income at these rates.
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Arkansas income tax rate
+  reference:
+  - title: Tax Legislation for 2022
+    href: https://www.arkansashouse.org/news/post/15098/tax-legislation-for-2022#:~:text=In%20the%20most%20recent%20special,rate%20is%20reduced%20to%205.5%25.
+  - title: State of Arkansas Estimated Tax Declaration Vouchers and Instructions for Tax Year 2023
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ar/tax/income/main.yaml:10]: Can you add a legal code reference? (Differs from tax legislation)

Might help with the special $87k-91.9k zone

**Comment by MaxGhenis** [comment]
Looks like you pulled in some unwanted changes, could you try `git pull upstream master`?

**Comment by MaxGhenis** [comment]
Please try git pull upstream master again and also resolve any conversations that you've addressed

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  label: Arkansas income tax rate
+  reference:
+  - title: Tax Legislation for 2022
+    href: https://www.arkansashouse.org/news/post/15098/tax-legislation-for-2022#:~:text=In%20the%20most%20recent%20special,rate%20is%20reduced%20to%205.5%25.
+  - title: State of Arkansas Estimated Tax Declaration Vouchers and Instructions for Tax Year 2023
+    href: https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/2023_Final_AR1000ES.pdf
+
+brackets:
+  - threshold:
+      2022-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:15]: need to add 2021 values and the 2021 source (tax form)

File: CHANGELOG.md
Comment (PavelMakarchuk) [inline CHANGELOG.md:1]: run `git pull upstream master` - not sure how these have been deleted but we need to add the code back

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:15]: need to add 2021 values and 2021 tax form

File: policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    value_type = int
+    entity = TaxUnit
+    label = "Arkansas income tax before credits"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/2023_Final_AR1000ES.pdf"
+    defined_for = StateCode.AR
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.ar.tax.income.rates.main
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py:16]:
```suggestion
        rate = parameters(period).gov.states.ar.tax.income.rates.main
```

File: policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from policyengine_us.model_api import *
+
+# import numpy as np
+
+
+class ar_income_tax_before_credits(Variable):
+    value_type = int
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py:7]:
```suggestion
    value_type = float
```

File: policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ar/tax/income/rates/ar_income_tax_before_credits.py:1]: need to add all variables as in [this PR ](https://github.com/PolicyEngine/policyengine-us/pull/2410)with the same code structure - just different state code in file names

File: CHANGELOG.md
Comment (PavelMakarchuk) [inline CHANGELOG.md:1]: run `git pull upstream master` to resolve the diff -> any time there are additional files that probably comes from files pulled while there are uncommitted changes

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
+description: Arkansas taxes personal taxable income at these rates.
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Arkansas income tax rate
+  reference:
+  - title: Tax Legislation for 2022
+    href: https://www.arkansashouse.org/news/post/15098/tax-legislation-for-2022#:~:text=In%20the%20most%20recent%20special,rate%20is%20reduced%20to%205.5%25.
+  - title: State of Arkansas Estimated Tax Declaration Vouchers and Instructions for Tax Year 2023
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:10]: don't need the 2023 estimation

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:10]: Please add the legislative reference - the legal code of Arkansas

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      2021-01-01: 0.034
+  - threshold:
+      2021-01-01: 22_900
+      2022-01-01: 24_300
+    rate:
+      2021-01-01: 0.05
+      2022-01-01: 0.049
+  - threshold:
+      2021-01-01: 38_500
+      #2022-01-01: 24_300
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:46]:
```suggestion
      2022-01-01: .inf
```

File: policyengine_us/variables/gov/states/ar/tax/income/ar_non_refundable_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ar/tax/income/ar_non_refundable_credits.py:1]: add another file called ar_income_tax_before_refundable_credits.py

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml (hunk 7 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:10]: https://casetext.com/statute/arkansas-code-of-1987/title-26-taxation/subtitle-5-state-taxes/chapter-51-income-taxes/subchapter-2-imposition-of-tax/section-26-51-201-effective-90-days-after-sine-die-adjournment-individuals-trusts-and-estates

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:1]: need changelog

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
+description: Arkansas taxes personal taxable income at these rates.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:1]:
```suggestion
description: Arkansas taxes personal income at these rates.
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Arkansas income tax rate
+  reference:
+  - title: Tax Legislation for 2022 Ark. Code § 26-51-201 (a)(1)(A)
+    href: https://casetext.com/statute/arkansas-code-of-1987/title-26-taxation/subtitle-5-state-taxes/chapter-51-income-taxes/subchapter-2-imposition-of-tax/section-26-51-201-effective-90-days-after-sine-die-adjournment-individuals-trusts-and-estates
+
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:10]: add 2022 and 2021 income tax form source

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  - title: Estimated Tax Declaration Vouchers and Instructions for Tax Year 2023
+    href: https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/2023_Final_AR1000ES.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:12]: add pages to tax form - we can also not use 2023 estimations - need 2022 tax form

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Arkansas income tax rate
+  reference:
+  - title: Tax Legislation for 2022 Ark. Code § 26-51-201 (a)(1)(A)
+    href: https://casetext.com/statute/arkansas-code-of-1987/title-26-taxation/subtitle-5-state-taxes/chapter-51-income-taxes/subchapter-2-imposition-of-tax/section-26-51-201-effective-90-days-after-sine-die-adjournment-individuals-trusts-and-estates
+
+  - title: Estimated Tax Declaration Vouchers and Instructions for Tax Year 2022
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:11]: this is still the estimated tax forms

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:11]: we can not use the estimation tax forms, we need to use the actual tax forms in all yaml files

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+  type: marginal_rate
+  threshold_unit: currency-USD
+  rate_unit: /1
+  label: Arkansas income tax rate
+  reference:
+  - title: Tax Legislation for 2022 Ark. Code § 26-51-201 (a)(1)(A)
+    href: https://casetext.com/statute/arkansas-code-of-1987/title-26-taxation/subtitle-5-state-taxes/chapter-51-income-taxes/subchapter-2-imposition-of-tax/section-26-51-201-effective-90-days-after-sine-die-adjournment-individuals-trusts-and-estates
+
+  - title: 2022 Indexed Tax Brackets
+    href: https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/TaxBrackets_2022.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:12]: need to add pdf paged in this format
https://www.dfa.arkansas.gov/images/uploads/incomeTaxOffice/TaxBrackets_2022.pdf#page=x

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ar/tax/income/rates/main.yaml:12]: all pdf sources and files

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**